src/utils/setup/on-create-page.js

```diff
@@ -0,0 +1,167 @@
+const memoizerific = require('memoizerific');
+const { getNestedValue } = require('../get-nested-value');
+const { getMetadata } = require('../get-metadata');
+
+const metadata = getMetadata();
+let stitchClient;
+
+const DEFAULT_FEATURED_HOME_SLUGS = [
+    '/how-to/nextjs-building-modern-applications',
+    '/how-to/python-starlette-stitch',
+    '/how-to/graphql-support-atlas-stitch',
+    '/quickstart/free-atlas-cluster',
+];
+
+const DEFAULT_FEATURED_LEARN_SLUGS = [
+    '/quickstart/java-setup-crud-operations',
+    '/how-to/golang-alexa-skills',
+    '/how-to/polymorphic-pattern',
+];
+
+// TODO: Verify field names with docsp
+const pageRouteToName = {
+    '/': 'home',
+    '/learn/': 'learn',
+};
+
+const requestStitch = async (functionName, ...args) =>
+    stitchClient.callFunction(functionName, [metadata, ...args]);
+const memoizedStitchRequest = memoizerific(10)(requestStitch);
+
+const getAllArticles = memoizerific(1)(async () => {
+    const documents = await memoizedStitchRequest('fetchDevhubMetadata', {});
+    // Ignore bad data, including links to the home page as an "article"
+    const filteredDocuments = documents.filter(d => {
+        const route = getNestedValue(['query_fields', 'slug'], d);
+        const title = getNestedValue(['query_fields', 'title'], d);
+        const image = getNestedValue(['query_fields', 'atf-image'], d);
+        return route !== '/' && !!title && !!image;
+    });
+    return filteredDocuments;
+});
+
+const getFeaturedArticles = (allArticles, featuredArticleSlugs) => {
+    const result = [];
+    featuredArticleSlugs.forEach((featuredSlug, i) => {
+        const newArticle = allArticles.find(
+            x => x.query_fields.slug === featuredSlug
+        );
+        if (newArticle) {
+            result.push(newArticle);
+        } else {
+            // This article was not found. pick an existing article and add it instead.
+            result.push(allArticles[i]);
+        }
+    });
+    return result;
+};
+
+const getLearnPageFilters = async () => {
+    const filters = {
```

Comment:
Unless I'm missing something we could just assign two `const`s for languages and products and return the and object at the end of the function like `return {languages, products}`. This way we don't have to keep enlarging a nested object or access nested properties in these loops.